src/data/set/intervals/unordered_interval.lean

```diff
@@ -27,97 +27,80 @@ make the notation available.
 
 open function order_dual (to_dual of_dual)
 
-namespace set
+variables {α β : Type*}
 
-section linear_order
-variables {α β : Type*} [linear_order α] [linear_order β] {f : α → β} {s : set α}
-  {a a₁ a₂ b b₁ b₂ c x : α}
+namespace set
+section lattice
+variables [lattice α] {a a₁ a₂ b b₁ b₂ c x : α}
 
-/-- `interval a b` is the set of elements lying between `a` and `b`, with `a` and `b` included. -/
-def interval (a b : α) := Icc (min a b) (max a b)
+/-- `interval a b` is the set of elements lying between `a` and `b`, with `a` and `b` included.
+Note that we define it more generally in a lattice as `set.Icc (a ⊓ b) (a ⊔ b)`. -/
```

Comment:
Can you update the module docstring too?

I also think it's worth providing a bit of motivation for this generalization; for instance, with the example of `interval (1, -1) (-1, 1)`, which gives the bounding box containing both points.

Reply:
Since it sounds like you want it for finset, a description for that case would be good too.

Reply:
Ah sorry I think you misunderstood what I want. I want `finset.interval`, a `finset`-valued version of `set.interval`.

Reply:
What type do you care about generalizing this too then? Is it the product type?

Reply:
I care about `finset (ι → ℤ)` and `finset (finset α)` (aka `finset (ι → Prop)`).

Reply:
Hmm... Maybe I do, but at any rate the explanation is the same as for `ι → ℤ`. It's the smallest bounding box, if you see `finset α` as an hypercube.

Reply:
Can you summarize the situation you need it in here? I think it would be good to give an example in the module doc that provides a little more insight than just "finsets are like products".